identity/common/src/main/java/io/camunda/identity/record/CamundaOidcUser.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.identity.record;
```

Comment:
I suggest structuring the code by the domain instead of class type, so for example having a `users` directory with all user related records and services instead of `service` and `records` directories. This structure should be more robust as the code grows. In Identity it became challenging to name classes in a single directory - especially records as these could include entities but also request objects and DTOs. What do you think?

Reply:
Sounds good, I've updated this structure now

Reply:
The idea of naming the packages with subdomain sounds good but I have some concerns here regarding how to find a subdomain that contains all relevant stuffs. For example when we named this package as user, then when we want to add group dtos we need to add a new package with one class only that's not looks nice, also for user roles and group roles and group users how we can proceed with this style?

Reply:
An example structure for roles: everything related to roles would live in a `roles` directory. This could be a subdirectory of  `users` or on the same level and would contain all DTOs and services to retrieve/manage roles. Working with subdirectories can give more granularity and a structure similar to api endpoints while having these on the same level should be easier for reusable parts like.
An example why I think that structuring by domain is a good idea is multitenancy. We would have a directory `tenants` containing all controllers, services, records, DTOs etc... and all of these can be annotated to only be included if multitenancy is activated. Otherwise the multitenancy annotation would be present on different classes in different directories and would be harder to follow